softlayer/resource_softlayer_bare_metal.go

```diff
@@ -76,100 +144,164 @@ func resourceSoftLayerBareMetal() *schema.Resource {
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
 			},
 
-			"public_subnet": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Computed: true,
+			// Monthly only
+			"process_key_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
 			},
 
-			"private_vlan_id": {
-				Type:     schema.TypeInt,
+			// Monthly only
+			"os_key_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"disk_key_names": {
+				Type:             schema.TypeList,
+				Optional:         true,
+				ForceNew:         true,
+				Elem:             &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"redundant_network": {
+				Type:     schema.TypeBool,
 				Optional: true,
+				Default:  false,
 				ForceNew: true,
-				Computed: true,
 			},
 
-			"private_subnet": {
-				Type:     schema.TypeString,
+			// Monthly only
+			"unbonded_network": {
+				Type:     schema.TypeBool,
 				Optional: true,
+				Default:  false,
 				ForceNew: true,
-				Computed: true,
 			},
 
-			"network_speed": {
+			// Monthly only
+			"public_bandwidth": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"memory": {
 				Type:     schema.TypeInt,
 				Optional: true,
-				Default:  100,
 				ForceNew: true,
-			},
-
-			"public_ipv4_address": {
-				Type:     schema.TypeString,
 				Computed: true,
 			},
 
-			"private_ipv4_address": {
-				Type:     schema.TypeString,
+			// Monthly only
+			"redundant_power_supply": {
+				Type:     schema.TypeBool,
+				Optional: true,
 				Computed: true,
 			},
 
-			"ssh_key_ids": {
+			// Monthly only
+			"storage_groups": {
 				Type:     schema.TypeList,
 				Optional: true,
-				Elem:     &schema.Schema{Type: schema.TypeInt},
 				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"array_type_id": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"hard_drives": {
+							Type:     schema.TypeList,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+							Required: true,
+						},
+						"array_size": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"partition_template_id": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+				DiffSuppressFunc: applyOnce,
 			},
 
-			"user_metadata": {
-				Type:     schema.TypeString,
+			// Quote based provisioning only
+			"quote_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
```

Comment:
I don't see why we would need both a quote id and a quote data source. Let's talk about that.